fvm/meter/meter.go

```diff
@@ -59,10 +64,15 @@ type Meter interface {
 	TotalMemoryEstimate() uint64
 	TotalMemoryLimit() uint64
 
-	// TODO move storage metering to here
-	// MeterStorageRead(byteSize uint) error
-	// MeterStorageWrite(byteSize uint) error
-	// TotalBytesReadFromStorage() int
-	// TotalBytesWroteToStorage() int
-	// TotalBytesOfStorageInteractions() int
+	// storage metering
+	MeterStorageRead(storageKey StorageInteractionKey, value flow.RegisterValue, enforceLimit bool) error
+	MeterStorageWrite(storageKey StorageInteractionKey, value flow.RegisterValue, enforceLimit bool) error
+	StorageUpdateSizeMap() MeteredStorageInteractionMap
```

Comment:
This should not be part of the interface.

Reply:
our current MergeMeter method takes `Meter` instead of `WeightedMeter`, thus to access WeightedMeter internals for merge purposes, we will need this interface to retrieve child data to merge. Similar for `ComputationIntensities()` and `MemoryIntensities()` above. Unless we do downcast from `*Meter` to `*WeightedMeter` in `MergeMeter()`, we can get rid of these interfaces. WDYT

Reply:
sounds good, I can do this deletion in a separate PR.